src/Glpi/Exception/Crud/UpdateFailedException.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Exception\Crud;
+
+use CommonDBTM;
+use Exception;
+use JsonException;
+
+use function Safe\json_encode;
+
+/**
+ * Unchecked exception that must be throw when an item update goes wrong in a place
+ * where the code don't expect it to happen.
+ *
+ * In the future, it could be thrown directly by the `update()` method of CommonDBTM
+ * as it is likely to be something we want to do by default (but would introduce BC breaks).
+ *
+ * In the meantime, a `safeUpdate()` method is available that will automatically
+ * wrap the `update` method and throw this exception on false returns to reduce
+ * boilerplate code.
+ */
+final class UpdateFailedException extends Exception
+{
+    public static function createFromItemAndInput(CommonDBTM $item, array $input): self
+    {
+        $base_message =
+            "Failed to update item of class '%s'." . PHP_EOL
+            . "Item fields: %s" . PHP_EOL
```

Comment:
Dumping inputs into the exception message/log file is a huge security risk.

Not sure how useful this is versus just having XDebug configured to start upon error or using breakpoints.